cve_bin_tool/nvd_api.py

```diff
@@ -93,35 +85,39 @@ async def get_nvd_params(
         """
         Initialize NVD request parameters
         """
-        if time_of_last_update:
-            # Fetch all cves from this date (even the updated ones)
-            self.params["modStartDate"] = self.convert_date_to_nvd_date(
-                time_of_last_update
-            )
-            self.logger.info(self.params["modStartDate"])
-            self.logger.info(time_of_last_update)
-            self.params["includeMatchStringChange"] = json.dumps(True)
-
-            # Check modified strings inside CVEs as well
-
         self.params["startIndex"] = 0
         self.params["resultsPerPage"] = self.page_size
 
         if not self.session:
-            connector = aiohttp.TCPConnector()
+            connector = aiohttp.TCPConnector(limit_per_host=19)
             self.session = RateLimiter(
                 aiohttp.ClientSession(connector=connector, trust_env=True)
             )
 
-        with Progress() as progress:
-            task = progress.add_task(
-                "Fetching metadata from NVD...", total=1, start=False
+        self.logger.debug("Fetching metadata from NVD...")
+        cve_count = await self.nvd_count_metadata(self.session)
+
+        if time_of_last_update:
+            # Fetch all the updated CVE entries from modified date. Taking 2 minute offset for updating cve entries
+            self.params["modStartDate"] = self.convert_date_to_nvd_date(
+                time_of_last_update - timedelta(minutes=2)
```

Comment:
I have subtracted 2 minutes time from (cvedb.db) timestamp considering the case NVD updates their database just about the time of populating the database